automation/service/workflow_step_args.go

```diff
@@ -0,0 +1,71 @@
+package service
+
+import (
+	"context"
+	"github.com/cortezaproject/corteza-server/automation/types"
+	"github.com/cortezaproject/corteza-server/pkg/expr"
+)
+
+type (
+	// argsProc is a helper to process arguments
+	//
+	// when initialised it evaluates expressions using given scope
+	// it provides methods to extract values from the result
+	argsProc struct {
+		expr   types.ExprSet
+		result *expr.Vars
+	}
+)
+
+func processArguments(ctx context.Context, expr []*types.Expr, scope *expr.Vars) (p *argsProc, err error) {
```

Comment:
Perhaps this would be cleanner

```go
func processArguments(ctx context.Context, scope *expr.Vars, expr ...*types.Expr)
```

Reply:
Primarily if no exprs are provided; is this even possible?

Reply:
a) not possible b) expressions are handled first (and are more important) and scope after that; and this dictates order of fn. args for me.